### PR TITLE
Salimans2016weight mmcinnestaylor

### PR DESCRIPTION
**Pull request by mmcinnestaylor**
Hey @liaojh1998! I realize the equations and figures are not yet in the md file. I am having difficulty getting the LaTeX to render correctly, and want to check in with Yue before I screenshot them all. I'll add the graphs from the results section soon.

### CODE REVIEW

File: _summaries/salimans2016weight_1.md [hunk 1]
```diff
@@ -1,19 +1,52 @@
...
+ * They then reparameterize the weight vectors, in terms of v, a k-dimensional vector, and g a scaling parameter  
+   * Equation 2 here  
+ * g may also be defined in terms of an exponential parameterization, where s is a log scale parameter learned by stochastic gradient descent  
+   * Scale Equation here  
+ * Training of the network used SGD, but the loss function now uses the reparameterizations above. Thus, the gradients in terms of those parameters are:  
+   * Equation 3 here
+   * Another formulation of the same gradients is:
+   * Equation 4 here
+   * This has the effect of scaling the weight gradient and projecting the gradient away from the current weight vector. The authors claim this brings the covariance matric of the gradient closer to the identity matrix, while also bebefiting optimization.
+ * The authors also propose the following data dependent initlalization of the network parameters when using Weight Normalization.
```
Comment (liaojh1998) [inline _summaries/salimans2016weight_1.md:27]: initlalization --> initialization

It would be a very good idea to mention why this is necessary, especially since batch normalization allows fixed feature outputs of 0 mean and unit variance whereas weight normalization doesn't. How does this initialization output an approximation of 0 mean and unit variance for feature output?

File: _summaries/salimans2016weight_1.md [hunk 2]
```diff
@@ -1,19 +1,52 @@
...
+ * g may also be defined in terms of an exponential parameterization, where s is a log scale parameter learned by stochastic gradient descent  
+   * Scale Equation here  
+ * Training of the network used SGD, but the loss function now uses the reparameterizations above. Thus, the gradients in terms of those parameters are:  
+   * Equation 3 here
+   * Another formulation of the same gradients is:
+   * Equation 4 here
+   * This has the effect of scaling the weight gradient and projecting the gradient away from the current weight vector. The authors claim this brings the covariance matric of the gradient closer to the identity matrix, while also bebefiting optimization.
+ * The authors also propose the following data dependent initlalization of the network parameters when using Weight Normalization. 
+   * Equations 5, 6
+   * The one downside to this initialization scheme is that it's application domains are limited to those where Batch Normalization performs well.
```
Comment (liaojh1998) [inline _summaries/salimans2016weight_1.md:29]: it's --> its

File: _summaries/salimans2016weight_1.md [hunk 3]
```diff
@@ -1,19 +1,52 @@
...
+ y = &phi;(w &middot; x + b)  
+ * They then reparameterize the weight vectors, in terms of v, a k-dimensional vector, and g a scaling parameter  
+   * Equation 2 here  
+ * g may also be defined in terms of an exponential parameterization, where s is a log scale parameter learned by stochastic gradient descent  
+   * Scale Equation here  
+ * Training of the network used SGD, but the loss function now uses the reparameterizations above. Thus, the gradients in terms of those parameters are:  
+   * Equation 3 here
+   * Another formulation of the same gradients is:
+   * Equation 4 here
+   * This has the effect of scaling the weight gradient and projecting the gradient away from the current weight vector. The authors claim this brings the covariance matric of the gradient closer to the identity matrix, while also bebefiting optimization.
```
Comment (liaojh1998) [inline _summaries/salimans2016weight_1.md:26]: bebefiting --> befitting

It might also be a good idea to mention that this is robust to different values of learning rates, and mention why that is so.

File: _summaries/salimans2016weight_1.md [hunk 4]
```diff
@@ -1,19 +1,52 @@
...
+ * results image 
+#### Generative Modelling - MNIST using DRAW  
+ * results image
+#### Reinforcement Learning - Deep-Q-Network  
+ * results image
+## Variants   
+The authors propose a hybrid method which utilizes Weight Normalization and a special version of Batch Normalization, which they call Mean-Only Batch Normalization.
+* Equations 7, 8
+* This method leads to the centering of the gradients which are back propogated, and is computationally less expensive than full batch norm.  
+* This method improves model accuracy at test time.
```
Comment (liaojh1998) [inline _summaries/salimans2016weight_1.md:46]: Maybe put Variants above the Performance so we don't get confused on the CIFAR-10 performance about mean-only bn.

File: _summaries/salimans2016weight_1.md [hunk 5]
```diff
@@ -1,19 +1,52 @@
...
+   * This has the effect of scaling the weight gradient and projecting the gradient away from the current weight vector. The authors claim this brings the covariance matric of the gradient closer to the identity matrix, while also bebefiting optimization.
+ * The authors also propose the following data dependent initlalization of the network parameters when using Weight Normalization. 
+   * Equations 5, 6
+   * The one downside to this initialization scheme is that it's application domains are limited to those where Batch Normalization performs well.      
+    
+  
+
+## Performance  
+#### Supervised Classification - CIFAR-10 using a modified ConvPool-CNN-C architecture
+* results image
```
Comment (liaojh1998) [inline _summaries/salimans2016weight_1.md:35]: Would be a good idea to mention why mean-only bn is good here: adding noise that's approximately Gaussian can be useful for regularizing the network.

File: _summaries/salimans2016weight_1.md [hunk 6]
```diff
@@ -1,19 +1,52 @@
...
+#### Generative Modelling - MNIST \& CIFAR-10 using a deep Convolutional VAE  
+ * results image 
+#### Generative Modelling - MNIST using DRAW  
+ * results image
+#### Reinforcement Learning - Deep-Q-Network  
+ * results image
```
Comment (liaojh1998) [inline _summaries/salimans2016weight_1.md:41]: Maybe just a general overview comment that:
* Batch normalization is 16% slower than standard parameterization
* Weight normalization is faster than batch normalization in the early stages of training
* Weight normalization usually outperformed standard parameterization

File: _summaries/salimans2016weight_1.md [hunk 7]
```diff
@@ -1,19 +1,52 @@
...
+* This method leads to the centering of the gradients which are back propogated, and is computationally less expensive than full batch norm.  
+* This method improves model accuracy at test time. 
 
 ## TL;DR
-* Three
-* Bullets
-* To highlight the core concepts
+* Weight Normalization reparameterizes neural network weight vectors, leading to faster training via improved optimization conditioning and faster SGD.
+* Weight Normalization is inspired by Batch Normalization, but is simpler to implement, computationally less expensive, and has broader applicability. 
+* A hybrid method, Mean-Only Batch Normalization, uses both Weight Norm and Batch Norm, and outperforms both methods in classification on CIFAR-10.
```
Comment (liaojh1998) [inline _summaries/salimans2016weight_1.md:51]: Mean-Only Batch Normalization with Weight Normalization

I don't think the mean-only bn is with wn. The authors also did separate experiments in CIFAR-10 where they used only mean-only bn, and then used mean-only bn combined with wn.

**Review by liaojh1998 — APPROVED** [review]
I'm going to take this draft as is without the edits you wanted to add and review it.

This is a good summary, and it's very nice and concise. The paper is quite math-heavy and may require lot's of understanding of batch normalization so props to you for getting through it. I think it may be good to expand a little on why some parts of the architecture are used. I also added some spelling error fixes but I think you'll already get those with your final draft. Good job!

**Comment by mmcinnestaylor** [comment]
@liaojh1998 Thank you for the very thorough review! I attempted to integrate most of your suggestions, and fill out the summary where needed.
